test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -10176,6 +10176,307 @@ BOOST_AUTO_TEST_CASE(address_overload_resolution)
 	BOOST_CHECK(callContractFunction("g()") == encodeArgs(u256(5)));
 }
 
+BOOST_AUTO_TEST_CASE(snark)
+{
+	char const* sourceCode = R"(
+	library Pairing {
```

Comment:
@chriseth if you want to license this non-GPL3 it would be useful adding a remark here before merging.

Reply:
Add a note to the gist, i.e. this copy of the code is GPL3, the gist is MIT.

Reply:
Why have two different licenses if they are the same code?

Reply:
To make it simpler.

Reply:
You mean adding a line saying "The snarks code is licensed under MIT" makes it complicated? :)

Reply:
Yes, because some other places in the code say "this is all GPL licensed except for this and that".

Reply:
Though that is usually a problem if there is a significant feature which has a more restrictive license. Using more permissive licensed code in a codebase with a more restrictive one doesn't really cause such a problem.